### ISSUE TITLE
Lingering test containers after `make test`

### ISSUE THREAD

**Issue by ividal** [mozilla-ai/lumigator#326]
**Describe the bug**
`make test` launches containers to enable integration tests. However, they are not torn down after.

**System Information (please complete the following information):**
 - OS: macOS Sonoma
 - Lumigator version: [7be2518](https://github.com/mozilla-ai/lumigator/commit/7be2518ec8c6bc59ab8463fc7c39aad078bbb386) 

**To Reproduce**
1. Clone repo
2. docker ps shows no containers
3. make test (requires 128GB available to build the images)
4. docker ps shows lingering containers

**Expected behavior**
No containes should be left there.

**Screenshots**
N/A

**Additional context**
Potentially one might want to only tear down those created during tests, but we can also agree that test should happen with no other lumigator containers running in the background anyway.

**Comment by aittalam** [comment]
Adding further context here:

- `make test` runs the following:
```
test:
    make start-lumigator-build
    cd lumigator/python/mzai/backend; SQLALCHEMY_DATABASE_URL=sqlite:///local.db uv run pytest
    cd lumigator/python/mzai/sdk; uv run pytest
    make stop-lumigator
```

There are a few checks missing:

- containers are started ignoring whether any container is already running (in the typical developer setting, they are already up in attached mode with `make local-up`)
- if anything breaks during tests, the last command (`make stop-lumigator`) is not run
- if everything works, all containers are teared down including those started with `make local-up`

Ideally what should happen is the following:

- check whether there are containers already running
- if so, run tests
- otherwise, (1) start containers, (2) run tests, (3) tear down containers

**Comment by ividal** [comment]
Fixed with #327
